scripts/infra

```diff
@@ -53,6 +53,11 @@ if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
 
         aws s3 cp "s3://${PFB_SETTINGS_BUCKET}/terraform/terraform.tfvars" "${PFB_SETTINGS_BUCKET}.tfvars"
 
+        # Workaround Git wanting everyone to be the same user
+        docker-compose run --rm \
+                    --entrypoint git \
+                    terraform config --global --add safe.directory /usr/local/src
```

Comment:
This is running with `docker-compose run --rm` and not writing to a persistent volume, so I think it sets that git config then immediately throws away the modified container, and the subsequent commands each run in a container of their own that doesn't have the config.

But actually, when I manually ran a Terraform container (`docker-compose run --entrypoint bash terraform`) and ran each of the commands in turn within the same container:
```
git config --global --add safe.directory /usr/local/src
rm -rf .terraform/ terraform.tfstate*
terraform init -backend-config="bucket=${PFB_SETTINGS_BUCKET}" -backend-config="key=terraform/state"
```
it still failed with the same error:
![image](https://user-images.githubusercontent.com/6598836/183498184-5b00cd1e-8966-4180-a1b8-519cf45bbf5c.png)

BUT If I do that then run the two recommended commands to whitelist the specific directories
```
git config --global --add safe.directory /usr/local/src/.terraform/modules/database
git config --global --add safe.directory /usr/local/src/.terraform/modules/vpc
```
then it works.

So yeah, I don't know how to get that git config to be available to the container that needs it.  It could be done in the Dockerfile, but that's way upstream. Also if the general `/usr/local/src` setting doesn't work and it needs to be specific to the modules, that would not be a suitable addition to the Dockerfile anyway.  I don't remember the nuances of docker-compose "volume" config well enough to know if there's a straightforward way to make that one file persist across containers.

Possibly we should go back to the user workaround...

Reply:
Went to new style mounting a gitconfig file to /root/.gitconfig workaround.